pandas/core/arrays/arrow/dtype.py

```diff
@@ -8,28 +8,42 @@
 
 from pandas.core.dtypes.base import StorageExtensionDtype
 
-from pandas.core.arrays.arrow import ArrowExtensionArray
-
 
 class ArrowDtype(StorageExtensionDtype):
     """
-    Base class for dtypes for BaseArrowArray subclasses.
+    Base class for dtypes for ArrowExtensionArray.
     Modeled after BaseMaskedDtype
     """
 
-    name: str
-    base = None
-    type: pa.DataType
-
     na_value = pa.NA
```

Comment:
This wasn't discussed in https://github.com/pandas-dev/pandas/pull/46774, but so I think this should be `pd.NA` instead of `pa.NA`. This is the actual object that users get when accessing a scalar missing value (not how it is stored under the hood), and so `pa.NA` has no useful / proper behaviour to serve as a user-facing object. 

For example StringDtype("pyarrow") is also using pd.NA.

Reply:
Hmm so if users want to actually want to access the underlying data, they need to convert the array to arrow first?

And this also means we need to ensure we are never building a new `pa.array/chunked_array` with `pd.NA`?

Reply:
Does pa.NA have the same semantics as pd.NA?

Reply:
No, and that's what I meant with "`pa.NA` has no useful / proper behaviour to serve as a user-facing object.". Basically `pa.NA` has no semantics whatsoever, apart from being a python object (and having a `as_py()` method that returns None, and being equal to itself). 

Note this is the case in general for the pyarrow scalar objects. They are all light wrappers around the C++ data, exposing a method to convert it to a proper python scalar, and for the rest having no functionality to be used as a scalar object in general context (eg the int scalar also can't be used as a number, for example adding numbers or comparing values).

Reply:
This extension array defines `__arrow_array__`, so if users want the underlying data they can call `pa.array(...)` (which seems reasonable to me)?

Gotcha, I see that `__getitem__` coerces scalars already with `as_py`, so given that I agree and I think it makes more sense to return `pd.NA` instead of `pa.NA`

Reply:
I do think we specifically want `pd.NA`, as the behaviour of `pd.NA` generally matches the behaviour of nulls in Arrow arrays (eg propagating in comparisons, instead of returning False)